pkg/kubelet/kubelet.go

```diff
@@ -2508,8 +2512,12 @@ func (kl *Kubelet) HandlePodAdditions(pods []*v1.Pod) {
 				for _, c := range podCopy.Spec.Containers {
 					allocatedResources, found := kl.statusManager.GetContainerResourceAllocation(string(pod.UID), c.Name)
 					if c.Resources.Requests != nil && found {
-						c.Resources.Requests[v1.ResourceCPU] = allocatedResources[v1.ResourceCPU]
-						c.Resources.Requests[v1.ResourceMemory] = allocatedResources[v1.ResourceMemory]
+						if _, ok := allocatedResources[v1.ResourceCPU]; ok {
+							c.Resources.Requests[v1.ResourceCPU] = allocatedResources[v1.ResourceCPU]
+						}
+						if _, ok := allocatedResources[v1.ResourceMemory]; ok {
+							c.Resources.Requests[v1.ResourceMemory] = allocatedResources[v1.ResourceMemory]
+						}
 					}
```

Comment:
It looks like it's a good time to wrap this  duplicated code into a function.

Reply:
+1. Looks good otherwise.

Reply:
Updated. PTAL @bart0sh @vinaykul